src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1045,6 +1049,28 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 	foreach(rangeTableCell, rangeTableList)
 	{
 		RangeTblEntry *rangeTableEntry = (RangeTblEntry *) lfirst(rangeTableCell);
+		bool referenceTable = false;
+
+		if (IsDistributedTable(rangeTableEntry->relid))
```

Comment:
> CREATE TEMP TABLE temp_nations(name text, key integer);
SELECT master_modify_multiple_shards('DELETE FROM multi_shard_modify_test USING temp_nations WHERE multi_shard_modify_test.t_value = temp_nations.key AND temp_nations.name = ''foobar'' ');
ERROR:  relation temp_nations is not distributed

I think I'd be an improvement in the error message :))